Separate the uninstall and install steps as subcommands

olimination opened the pull request:
- Refactored/Renamed several names for better understanding
- Used separate uninstall and install configurations, because the uninstall case may rely on another set of artifacts or packages
- Updated deployment-configuration.yaml
- Updated README

Commit:
Separate the uninstall and install steps as subcommands

- Refactored/Renamed several names for better understanding
- Used separate uninstall and install configurations, because the uninstall case may rely on another set of artifacts or packages
- Updated deployment-configuration.yaml
- Updated README

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/unic/sledge?pullRequest=44) <br/>All committers have signed the CLA.

olimination commented:
Input from @pawel-o: Will simplify deployment-configuration.yaml such we don't need an `uninstallationDef` section. But we will update the Deployment docs, such you simply use the previous delivery package to uninstall the old release and then you use the new delivery package to do the installation.

Commit:
Incorporate pull request review inputs, simplify deployment-configuration.yaml

- Removed uinstallationDef and installationDef and using only one deploymentDef
- Uninstallation can be handled by using simply the old version of a release package, so no need to have uninstallationDef

Commit:
Remove build-scan plugin, because it is not used currently
